samples-generator/lib/createBuildingsTile.js

```diff
@@ -91,24 +104,57 @@ function createBuildingsTile(options) {
         featureTableJson.RTC_CENTER = Cartesian3.pack(batchedMesh.getCenter(), new Array(3));
     }
 
-    return createGltf({
+    var gltfOptions = {
         mesh : batchedMesh,
         useBatchIds : useBatchIds,
         relativeToCenter : relativeToCenter,
-        deprecated : deprecated1 || deprecated2
-    }).then(function(glb) {
-        var b3dm = createB3dm({
-            glb : glb,
-            featureTableJson : featureTableJson,
-            batchTableJson : batchTableJson,
-            batchTableBinary : batchTableBinary,
-            deprecated1 : deprecated1,
-            deprecated2 : deprecated2
+        deprecated : deprecated1 || deprecated2,
+        use3dTilesNext : use3dTilesNext
+    };
+
+    var gltf = createGltf(gltfOptions);
+
+    var b3dmOptions = {
+        featureTableJson : featureTableJson,
+        batchTableJson : batchTableJson,
+        batchTableBinary : batchTableBinary,
+        batchTableJsonAndBinary : batchTableJsonAndBinary,
+        deprecated1 : deprecated1,
+        deprecated2 : deprecated2,
+    };
+
+    var binary = defined(b3dmOptions.batchTableBinary) ? b3dmOptions.batchTableJsonAndBinary.binary : undefined;
+
+    // Don't add the batch table extension if there is no batchTableJson (e.g in the case of `createBatchedWithoutBatchTable`)
+    if (use3dTilesNext && defined(b3dmOptions.batchTableJson)) {
+        gltf = create3dtilesBatchTableExt(gltf, b3dmOptions.batchTableJson, binary);
+
+        // Make the RTC_CENTER translation the child of the rotationMatrix rootNode
+        gltf = injectTranslationNodeToGltf(gltf, 'RTC_CENTER', featureTableJson.RTC_CENTER, 0);
```

Comment:
`injectTranslationNodeToGltf` shouldn't be called if `featureTableJson.RTC_CENTER` is undefined. `BatchedWGS84` is one case like that.